mkt/developers/tasks.py

```diff
@@ -63,17 +64,15 @@ def validator(upload_id, **kw):
         return
 
     temp_path = None
-    # Make a copy of the file if it's a packaged app since we can't assume the
+    # Make a copy of the file since we can't assume the
     # uploaded file is on the local filesystem.
-    if upload.name.endswith('.zip'):
```

Comment:
I didn't like the `endswith('.zip')` check, but I'm not entirely sure it's a good idea to even copy hosted apps manifests. The validator does support hosted apps URLs just fine, I'm not sure how it works when you feed it a local hosted app manifest